packages/dds/merge-tree/src/test/client.spec.ts

```diff
@@ -207,6 +208,406 @@ describe("TestClient", () => {
 		});
 	});
 
+	describe(".searchForMarker", () => {
+		it("Should not return tile when searching past the end of a string length 1", () => {
+			client.insertMarkerLocal(0, ReferenceType.Tile, {
+				markerId: "marker",
+				referenceTileLabels: ["Eop"],
+			});
+
+			assert.equal(client.getLength(), 1);
+			const foundTile = client.searchForMarker(client.getLength(), "Eop", true);
+
+			assert.equal(typeof foundTile, "undefined", "Returned tile should be undefined.");
+		});
+
+		it("Should not return tile when searching before the start of a string length 1", () => {
+			client.insertMarkerLocal(0, ReferenceType.Tile, {
+				markerId: "marker",
+				referenceTileLabels: ["Eop"],
+			});
+
+			assert.equal(client.getLength(), 1);
+
+			const foundTile = client.searchForMarker(-1, "Eop", false);
+
+			assert.equal(typeof foundTile, "undefined", "Returned tile should be undefined.");
+		});
+
+		it("Should not return tile when searching past the end of a string length > 1", () => {
+			client.insertMarkerLocal(0, ReferenceType.Tile, {
+				markerId: "marker",
+				referenceTileLabels: ["Eop"],
+			});
+			client.insertTextLocal(0, "abc");
+
+			assert.equal(client.getLength(), 4);
+
+			const foundTile = client.searchForMarker(client.getLength(), "Eop", true);
+
+			assert.equal(typeof foundTile, "undefined", "Returned tile should be undefined.");
+		});
+
+		it("Should not return tile when searching before the start of a string length > 1", () => {
+			client.insertTextLocal(0, "abc");
+			client.insertMarkerLocal(0, ReferenceType.Tile, {
+				markerId: "marker",
+				referenceTileLabels: ["Eop"],
+			});
+
+			assert.equal(client.getLength(), 4);
+
+			const foundTile = client.searchForMarker(-1, "Eop", false);
+
+			assert.equal(typeof foundTile, "undefined", "Returned tile should be undefined.");
+		});
+
+		it("Should not return tile at the search position in either direction", () => {
+			client.insertTextLocal(0, "abcdefg");
+			client.insertMarkerLocal(4, ReferenceType.Tile, {
+				markerId: "marker",
+				referenceTielLabels: ["Eop"],
+			});
+			assert.equal(client.getLength(), 8);
+
+			const tile1 = client.searchForMarker(4, "Eop", true);
+
+			assert.equal(typeof tile1, "undefined", "Returned tile should be undefined.");
```

Comment:
nit: no need to `typeof`:

```suggestion
			assert.equal(tile1, undefined, "Returned tile should be undefined.");
```